R/generic.R

```diff
@@ -133,13 +133,37 @@ check_generic <- function(fun) {
     stop("`fun` must contain a call to `R7_dispatch()`", call. = FALSE)
   }
 }
+
+#' Find a call in expression
+#'
+#' Find a call (namespaced or plain) in an language object.
+#'
+#' @param x An language object in which to search for a particular function.
+#' @param name A quoted function name to search for. Both the plain function
+#'   name and qualified function name from the package namespace are considered.
+#' @return The `call` object if found, or `NULL` otherwise.
+#' @keywords internal
+#'
 find_call <- function(x, name) {
+  call_names <- list(ns_call(name), name)
+  find_one_of_calls(x, call_names)
+}
+
+ns_call <- function(name) {
+  call("::", as.symbol(packageName()), name)
+}
+
+find_one_of_calls <- function(x, call_names) {
   if (is.call(x)) {
-    if (identical(x[[1]], name)) {
-      return(x)
-    } else if (length(x) > 1) {
+    for (call_name in call_names) {
```

Comment:
This ended up being a bit more involved because `find_call` doesn't recurse through the call expressions themselves. This could move to a separate function, but I left it in-line for now.

```r
    if (!is.null(ns) && length(x[[1]]) == 3 && identical(x[[1]][[1]], quote(`::`))) {
      if (identical(x[[1]][[2]], as.symbol(ns)) && identical(x[[1]][[3]], name)) {
        return(x)
      }
    }
```

Reply:
Sorry, now that I've reminded myself how this function actually works, I think your initial implementation was cleaner 😞

Reply:
There are elements I like from both implementations. I'll see if I can come up with a nice blend.

Also - Great R7 talk! :1st_place_medal:

Reply:
Borrowing ideas from both implementations, I'm much happier with where it ended up. I like the specificity of the above solution, but broke it off into a separate function. The primary logic now reads quite cleanly:

```r
    if (is_ns_call(x, name, ns) || identical(x[[1]], name)) {
      return(x)
    }
```

Reply:
Ah, nice! I did a little refactoring and reduced the docs a bit, focussing more on the arguments types, because I think that's where confusion is more likely to arise for other devs.